### ISSUE TITLE
Switch to `Backblaze/blazer` for the B2 backend

### ISSUE THREAD

**Issue by MichaelEischer** [restic/restic#4148]
According to https://github.com/kurin/blazer/pull/84#issuecomment-1382526369 the new upstream is https://github.com/Backblaze/blazer . We have to wait until https://github.com/Backblaze/blazer/pull/2 is merged before we can switch.

**Comment by nocha1n** [comment]
This appears merged in [v0.6.0](https://github.com/Backblaze/blazer/releases/tag/v0.6.0).

**Comment by MichaelEischer** [comment]
Oh indeed. Anyone up for a PR?

**Comment by adamantike** [comment]
When trying to move to `Backblaze/blazer`, it fails with:
```
package github.com/restic/restic/cmd/restic
        imports github.com/restic/restic/internal/backend/b2
        imports github.com/Backblaze/blazer/b2
        imports github.com/Backblaze/blazer/base
        /home/mike/.go/pkg/mod/github.com/!backblaze/blazer@v0.6.0/base/base.go:40:2: use of internal package github.com/kurin/blazer/internal/b2types not allowed
package github.com/restic/restic/cmd/restic
        imports github.com/restic/restic/internal/backend/b2
        imports github.com/Backblaze/blazer/b2
        imports github.com/Backblaze/blazer/base
        /home/mike/.go/pkg/mod/github.com/!backblaze/blazer@v0.6.0/base/base.go:41:2: use of internal package github.com/kurin/blazer/internal/blog not allowed
```

https://github.com/Backblaze/blazer/pull/6 fixes the issue, so I will give it another try when that PR gets merged and released.

**Comment by MichaelEischer** [comment]
Thanks for giving it a try. The fix has been released as part of 0.6.1.
